### PR TITLE
ArrowFormatFactoryTest unit test added

### PR DESCRIPTION
**Pull request by Sridhar-Durgam2022**

### CODE REVIEW

File: src/main/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchema.java [hunk 1]
```diff
@@ -35,49 +36,78 @@
...
-
   private VectorSchemaRoot root;
   private VectorLoader loader;
   List<FieldVector> vectors = new ArrayList<>();
   private Schema schema;
   private final Class<T> recordClazz;
 
+  private String schemaJsonString;
+
+  private List<String> selectedFields;
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchema.java]: Removed unused variable

File: src/main/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchema.java [hunk 2]
```diff
@@ -35,49 +36,78 @@
...
+
+  private Schema readSessionSchema;
+
+  private Schema rowTypeSchema;
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchema.java]: Removed unused variable

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+      long dateTime = 0;
+      if (!(object.toString().length() == 26)) {
+        final DateTime value =
+            DateTime.parse(object.toString(), ISODateTimeFormat.dateTimeParser().withZoneUTC());
+
+        dateTime = value.toDate().getTime();
+      } else if (object.toString().length() == 26) {
+        DateTime value =
+            DateTime.parse(object.toString(), ISODateTimeFormat.dateTimeParser().withZoneUTC());
+        dateTime = value.toDate().getTime();
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:330]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/test/java/com/google/cloud/flink/bigquery/ArrowFormatFactoryTest.java [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+  public static final int DEFAULT_PARALLELISM = 10;
+  public static final String FLINK_VERSION = "1.11.0";
+  ImmutableMap<String, String> defaultOptions = ImmutableMap.of("table", "dataset.table");
+
+  public ArrowFormatFactoryTest() {
+
+    StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+    flinkTableEnv = StreamTableEnvironment.create(env);
+
+    bigqueryReadTable = "q-gcp-6750-pso-gs-flink-22-01.wordcount_dataset.wordcount_output";
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/ArrowFormatFactoryTest.java:70]: please use environment variable supplied project and the test dataset

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]
* Do not test using toString()

File: src/test/java/com/google/cloud/flink/bigquery/ArrowFormatFactoryTest.java (hunk 4 above)
Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/ArrowFormatFactoryTest.java]: added dummy project details : project.dataset.table

File: src/main/java/com/google/cloud/flink/bigquery/ArrowRowDataDeserializationSchema.java [hunk 5]
```diff
@@ -58,8 +64,37 @@ public ArrowRowDataDeserializationSchema(
...
     this.runtimeConverter =
-        ArrowToRowDataConverters.createRowConverter(
-            rowType, readSessionFieldNames, selectedFieldList);
+        ArrowToRowDataConverters.createRowConverter(rowType, readSessionFieldNames);
+  }
+
+  /**
+   * @param rowType
+   * @param selectedFieldList
+   * @param arrowFieldList
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/ArrowRowDataDeserializationSchema.java]: removed

File: src/main/java/com/google/cloud/flink/bigquery/ArrowRowDataDeserializationSchema.java [hunk 6]
```diff
@@ -58,8 +64,37 @@ public ArrowRowDataDeserializationSchema(
     this.arrowSchemaJson = arrowSchema.toJson().toString();
     this.nestedSchema = ArrowDeserializationSchema.forGeneric(arrowSchemaJson, typeInfo);
     this.runtimeConverter =
-        ArrowToRowDataConverters.createRowConverter(
-            rowType, readSessionFieldNames, selectedFieldList);
+        ArrowToRowDataConverters.createRowConverter(rowType, readSessionFieldNames);
+  }
+
+  /**
+   * @param rowType
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/ArrowRowDataDeserializationSchema.java]: removed

File: src/main/java/com/google/cloud/flink/bigquery/ArrowRowDataDeserializationSchema.java [hunk 7]
```diff
@@ -58,8 +64,37 @@ public ArrowRowDataDeserializationSchema(
...
+   * @param selectedFieldList
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/ArrowRowDataDeserializationSchema.java]: removed

**Pull request closed without merging**